File: EPSILON/SAMPLE FILE/E-publishing/Tools/Altova/Altova/XMLSpy2006/spl/cpp/AltovaText/Record.cpp
[call GenerateFileHeader("Record.cpp")]
#include "stdafx.h"
#include "Record.h"

namespace altova
{
namespace text
{
namespace tablelike
{


CRecord::CRecord(size_t fieldcount)
:	m_FieldCount(fieldcount)
{
	m_Fields= new tstring\[m_FieldCount\];
}

CRecord::CRecord(const CRecord& rhs)
{
	m_FieldCount = rhs.m_FieldCount;
	m_Fields = new tstring\[m_FieldCount\];
	for (size_t i = 0; i < m_FieldCount; ++i)
		m_Fields\[i\] = rhs.m_Fields\[i\];
}

CRecord::~CRecord()
{
	delete \[\] m_Fields;
}

void CRecord::Assign(const CRecord& rhs)
{
	delete \[\] m_Fields;
	m_FieldCount = rhs.m_FieldCount;
	m_Fields = new tstring\[m_FieldCount\];
	for (size_t i = 0; i < m_FieldCount; ++i)
		m_Fields\[i\] = rhs.m_Fields\[i\];
}

bool CRecord::HasFieldAt(size_t index) const
{
	return (index < m_FieldCount);
}

const tstring& CRecord::GetFieldAt(size_t index) const
{
	static const tstring empty = _T("");
	if (0 <= index && index < m_FieldCount)
		return m_Fields\[index\];
	else
		return empty;
}

void CRecord::SetFieldAt(size_t index, const tstring& rhs)
{
	if (0 <= index && index < m_FieldCount)
		m_Fields\[index\] = rhs;
}

} // namespace tablelike
} // namespace text
} // namespace altova

File: EPSILON/SAMPLE FILE/E-publishing/Tools/Altova/Altova/XMLSpy2006/spl/java/altova/text/tablelike/Record.java
[call GenerateFileHeader("Record.java")]
package com.altova.text.tablelike;

import java.util.Arrays;

public class Record {
    private String\[\] m_Fields = null;

    Record(String\[\] fields) {
        m_Fields = new String\[fields.length\];
        for (int i = 0; i < fields.length; ++i)
            m_Fields\[i\] = fields\[i\];
    }

    Record(StringList fields) {
        m_Fields = new String\[fields.size()\];
        fields.toArray(m_Fields);
    }

    public Record(int fieldcount) {
        m_Fields = new String\[fieldcount\];
    }

    public Record(Record rhs) {
        m_Fields = new String\[rhs.m_Fields.length\];
        for (int i = 0; i < m_Fields.length; ++i)
            m_Fields\[i\] = rhs.m_Fields\[i\];
    }

    public int size() {
        return m_Fields.length;
    }

    public String getAt(int index) {
        if (index >= m_Fields.length)
            return null;
        return m_Fields\[index\];
    }

    public void setAt(int index, String rhs) {
        m_Fields\[index\] = rhs;
    }
}
